src/analyze/analyze.c

```diff
@@ -2423,6 +2426,13 @@ static int parse_argv(int argc, char *argv[]) {
 
                         break;
 
+                case 'U':
+                        r = unit_name_mangle(optarg, UNIT_NAME_MANGLE_WARN, &arg_unit);
```

Comment:
hmm, if users specify -U twice now, you'll leak memory. Hence, please store result of unit_name_mangle() in temp var first. then, on success, free the old arg_unit, and assign the new. (`free_and_replace()`)